### PR TITLE
[chore] use distributions.yaml in mdatagen

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Use the distributions.yaml introduced by #29697 to get the list of distributions.

### CODE REVIEW

**Review by songy23 — COMMENTED** [review]
> Error: statusdata.go:14:12: pattern .distributions.yaml: no matching files found

File: .gitignore [hunk 1]
```diff
@@ -39,3 +39,5 @@ integration-coverage.html
 go.work*
 
 /result
+# File copied into mdatagen to be embedded
+cmd/mdatagen/.distributions.yaml
```
Comment (bogdandrutu) [inline .gitignore:43]: Why copy/ignore vs not having it here in from the beginning?

Comment (atoulme) [reply .gitignore:43]: the file was already present in the root of the repository. There is a go:generate trick used to copy the file in the work folder:
`//go:generate cp -r ../../distributions.yaml .distributions.yaml`

This trick doesn't work with `go install` though. Yes, we could just keep it around in git instead, you're right.

Comment (atoulme) [reply .gitignore]: OK, this should be good now.

**Review by songy23 — APPROVED** [review]

File: .gitignore [hunk 2]
```diff
@@ -38,4 +38,4 @@ integration-coverage.html
 
 go.work*
 
-/result
+/result
```
Comment (bogdandrutu) [inline .gitignore:41]: Why this change?

**Comment by bogdandrutu** [comment]
Also I am confused about the intent of the PR in general, why not using githubgen for this?

File: .gitignore (hunk 2 above)
Comment (atoulme) [reply .gitignore:41]: newline got nicked, sorry. Fixing.

**Comment by atoulme** [comment]
> Also I am confused about the intent of the PR in general, why not using githubgen for this? Can you please clarify why we need this feature.

We use the list of distributions to render the list with hyperlinks in the README of components presently.
We have moved the list of distributions to a centralized yaml file at the root of the repository, which is used to generate a distribution report under report/distributions.

This closes the loop on the change, by adopting the distribution data from the yaml file as the source of truth for component distributions. This allows us to maintain distributions by editing distributions.yaml and see it reflected in all READMEs and distribution reports. See https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/29697 for context and https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/28628 for the original motivation.

**Comment by atoulme** [comment]
@dmitryax please review as codeowner?

**Comment by dmitryax** [comment]
How this change will play with the fact that mdatagen is moved to core? I think we need to start planning the deprecation and removal of this package from contrib

**Comment by atoulme** [comment]
you tell me :)
